frog/imports/client/AccountModal/SignUp.js

```diff
@@ -0,0 +1,210 @@
+
+    return (
+      <Container component="main" maxWidth="xs">
+        <CssBaseline />
+        <div className={classes.paper}>
+          <Avatar className={classes.avatar}>
+            <LockOutlinedIcon />
+          </Avatar>
+          <Typography component="h1" variant="h5">
+            Create an account with FROG
+          </Typography>
+          <form
+            className={classes.form}
+            onSubmit={e => this.handleSubmit(e)}
+            noValidate
+          >
+            <Grid container spacing={2}>
+              <Grid item xs={12}>
+                <TextField
+                  autoComplete="fname"
+                  fullWidth
+                  name="displayName"
+                  error={this.state.formErrors.displayName !== ''}
+                  variant="outlined"
+                  required
+                  helperText={this.state.formErrors.displayName}
+                  id="displayName"
+                  label="Display Name"
+                  onChange={e => this.handleChange(e, 'displayName')}
+                  autoFocus
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  error={this.state.formErrors.email !== ''}
+                  required
+                  fullWidth
+                  id="email"
+                  label="Email Address"
+                  name="email"
+                  helperText={this.state.formErrors.email}
+                  onChange={e => this.handleChange(e, 'email')}
+                  autoComplete="email"
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  variant="outlined"
+                  error={this.state.formErrors.password !== ''}
+                  required
+                  fullWidth
+                  name="password"
+                  label="Password"
+                  type="password"
+                  id="password"
+                  helperText={this.state.formErrors.password}
+                  onChange={e => this.handleChange(e, 'password')}
+                  autoComplete="current-password"
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <Button
+                  type="submit"
+                  fullWidth
+                  variant="contained"
+                  color="primary"
+                  className={classes.submit}
+                >
+                  Sign Up
+                </Button>
+              </Grid>
+            </Grid>
+
+            <Grid container justify="flex-end">
+              <Grid item>
+                <Link href="#" variant="body2">
+                  Already have an account? Sign in
+                </Link>
+              </Grid>
+            </Grid>
+          </form>
+        </div>
+      </Container>
+    );
+  }
+}
+
+export default withStyles(styles)(SignUp);
```

Comment:
We should a hook, bonus is that `classes` gets automatic type inference :tada:

Reply:
Discussed and closed